[Patrick Navin Thomas] iP

pastchum opened the pull request:
### MIKI
_Your DAWG Chatbot_

Get yourself a real dawg that helps you settle all your tasks and what not. 

- command-based
- easy to learn
- no fuss

All you need to do is,
1. [download](https://github.com/pastchum/ip).
2. open.
3. say your tasks.
4. let MIKI do the job for you

**Settle Kettle** bro.

> 
Everyone loves MIKI. 🐕

Features:
- [ ]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

Here's the main method for the top DAWG assistant:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

pastchum committed:
changed name

pastchum committed:
changed the test case

pastchum committed:
changed greeting message

pastchum committed:
test case fixed

pastchum committed:
Merge pull request #1 from pastchum/level-0

Level 0

pastchum committed:
echoing feature works

pastchum committed:
minor tweaks to UI

pastchum committed:
fixed test case

pastchum committed:
Merge pull request #2 from pastchum/level-1

Level 1

pastchum committed:
create task class

pastchum committed:
add tasks function added

pastchum committed:
fixed test cases

pastchum committed:
Merge pull request #3 from pastchum/level-2

Level 2

pastchum committed:
task class now has completion tracking

pastchum committed:
added mark and unmark tasks

pastchum committed:
fixed marking already marked task

pastchum committed:
test case updated

pastchum committed:
Merge pull request #4 from pastchum/level-3

Level 3

pastchum committed:
Task now abstract, Task classes created

pastchum committed:
logic added

pastchum committed:
event out of bounds fix

pastchum committed:
fixed messages

pastchum committed:
test case updated

pastchum committed:
Merge pull request #5 from pastchum/level-4

Level 4

pastchum committed:
created exception classes

pastchum committed:
Exception logic handles

pastchum committed:
new exception for too many tasks

pastchum committed:
Merge pull request #6 from pastchum/level-5

Level 5

pastchum committed:
added delete command and exception

pastchum committed:
bug fixes

pastchum committed:
Merge pull request #7 from pastchum/level-6

Level 6

pastchum committed:
Create Storage class

pastchum committed:
Create TaskList class

pastchum committed:
Make Miki Class OOP

pastchum committed:
Fix Storage file creation

pastchum committed:
Fix Storage write

pastchum committed:
Fix Storage save task

pastchum committed:
Add Ui class

pastchum committed:
Fix Ui Error message

pastchum committed:
Move Scanner to Ui

pastchum committed:
Add Command package

pastchum committed:
Add Parse package

pastchum committed:
Fix Storage Task save error

pastchum committed:
Update Text-ui-test Expected

pastchum committed:
Merge pull request #8 from pastchum/branch-level-7

Branch level 7

pastchum committed:
Fix import statements

pastchum committed:
Add Date parsing

pastchum committed:
Fix Event date logic

pastchum committed:
Merge pull request #9 from pastchum/branch-level-8

Branch level 8

pastchum committed:
Fix Task LocalDateTime

pastchum committed:
Fix LocalDateTime formatting

pastchum committed:
Add Find function

pastchum committed:
Fix Java Coding Standard

pastchum committed:
Merge pull request #10 from nus-cs2103-AY2425S2/add-gradle-support

Add gradle support

nja99 commented on src/main/java/miki/task/TaskList.java:
```diff
@@ -0,0 +1,36 @@
+        tasks.remove(index);
+    }
+
+    public int size() {
+        return tasks.size();
+    }
```
Perhaps getSize() would be a more descriptive function name?

nja99 commented on src/main/java/miki/Miki.java:
```diff
@@ -0,0 +1,58 @@
+import miki.ui.Ui;
+
+public class Miki {
+    private static TaskList tasks;
```
Would it be better to name the variable as tasksList as to not get confused with task?
I noticed the same issue in several other places too.

nja99 commented on src/main/java/miki/task/Task.java:
```diff
@@ -0,0 +1,62 @@
+import miki.exception.MikiException;
+
+public abstract class Task {
+    private String description;
+    private boolean isCompleted;
+    private TaskType taskType;
```
I really like how you've named the variables here, they make the code very readable and intuitive

nja99 commented on src/main/java/miki/task/Task.java:
```diff
@@ -0,0 +1,62 @@
+        }
+    }
+
+    public void toggleCompletion() {
```
While toggleCompletion accurately reflects this function's behaviour, it might be better to have distinct function for clarity.

nja99 reviewed (commented):
Generally, the naming throughout the codebase was well done apart from minor infractions.

pastchum committed:
Add Task Tests

pastchum committed:
Merge pull request #11 from pastchum/branch-A-CodingStandard

Branch a coding standard

pastchum committed:
Fix Find Task logic

pastchum committed:
Merge pull request #12 from pastchum/branch-level-9

Branch Level 9

pastchum committed:
Add Headers for Task

pastchum committed:
Add Ui Headers

pastchum committed:
Add Storage Headers

pastchum committed:
Add Parser Headers

pastchum committed:
Neaten Parser logic

pastchum committed:
Add Exception Headers

pastchum committed:
Add Command Headers

pastchum committed:
Add Miki Headers

pastchum committed:
Merge pull request #13 from pastchum/branch-A-JavaDoc

Branch a java doc

pastchum committed:
Add GUI

pastchum committed:
Change User and Miki image

pastchum committed:
Merge pull request #14 from pastchum/branch-level-10

Branch level 10

pastchum committed:
Add CheckStyle configuration files

pastchum committed:
Fix CheckStyle errors

pastchum committed:
Merge pull request #15 from pastchum/branch-A-CheckStyle

Branch a check style

pastchum committed:
Add assertion

pastchum committed:
Add Assertions

Command Classes: Add Assertions

Currently, Command classes and all their execute functions do not have error checks before throwing out an error.

This can lead to issues for the tasks pulled from the TaskList tasklist in the event that it returns a null value.

Let's,

update the error checks to include assertions
add a string to briefly explain what the assertion error may be
This is done to prevent NullPointerExceptions when running Command.execute().

pastchum committed:
Rename AddTodoCommand.java to AddToDoCommand.java

pastchum committed:
Add Streams

pastchum committed:
Fix Test checkstyle error

pastchum committed:
Merge pull request #18 from pastchum/branch-A-Streams

TaskList: uses Streams

TaskList uses for-loop.

Streams allows for neater code when printing every
Task in the TaskList

Let's,

update the for-loop to a stream

pastchum committed:
Create gradle.yml

pastchum committed:
Fix Ui method headings

pastchum committed:
Refactor Parser

pastchum committed:
Add Parser SLAP

pastchum committed:
Merge pull request #19 from pastchum/branch-A-CI

Add CI

pastchum committed:
Merge pull request #20 from pastchum/branch-A-CodeQuality

Branch a code quality

pastchum committed:
Update workflows

pastchum committed:
Add Tag feature

pastchum committed:
Fix parse date bug

pastchum committed:
Fix CheckStyle error

pastchum committed:
Add Ui test

pastchum committed:
Merge pull request #21 from pastchum/branch-BCD-Extension

Branch bcd extension

pastchum committed:
Add Command and Parse Test

pastchum committed:
Merge pull request #22 from pastchum/branch-A-moreTesting

Add Command and Parse Test

pastchum committed:
Update README.md

pastchum committed:
Update README

pastchum committed:
Merge pull request #23 from pastchum/branch-A-UserGuide

Branch a user guide

pastchum committed:
Create jekyll-gh-pages.yml

pastchum committed:
Merge pull request #24 from pastchum/branch-A-UserGuidePage

Create jekyll-gh-pages.yml

pastchum committed:
Update GUI

pastchum committed:
Fix ToDo description bug

pastchum committed:
Fix Save Task bug

pastchum committed:
Update README

pastchum committed:
Update README

pastchum committed:
Merge pull request #25 from pastchum/branch-A-BetterGUI

Branch a better g UI

pastchum committed:
Remove Last Expected Output

pastchum committed:
Update UI showcase

pastchum committed:
Merge pull request #26 from pastchum/branch-A-BetterGUI

Branch a better g UI

pastchum committed:
Update README

pastchum committed:
Merge pull request #27 from pastchum/branch-A-BetterGUI

Update README

pastchum committed:
Fix Parser Command bug

pastchum committed:
Fix README image path bug

pastchum committed:
Fix Parse Storage bug

pastchum committed:
Clear Tasks

pastchum committed:
Fix Storage bug

pastchum committed:
Fix Exit Feature

pastchum committed:
Merge pull request #28 from pastchum/branch-A-BetterGUI

Branch a better g UI

pastchum committed:
Fix CheckStyle

pastchum committed:
Merge pull request #29 from pastchum/branch-A-BetterGUI

Fix CheckStyle

pastchum committed:
Update README

pastchum committed:
Merge pull request #30 from pastchum/branch-A-BetterGUI

Update README
